File: medium/container-with-most-water.kt
class Solution {
    fun maxArea(height: IntArray): Int {
        // The first submission -> 620ms
        var l = 0;
        var r = height.size - 1;
        var max = 0;
        var curr = 0;

        while (l < r) {
            if (height[l] < height[r]) {
                curr = height[l] * (r - l);
                l++;
            } else {
                curr = height[r] * (r - l);
                r--;
            }
            if (curr > max) {
                max = curr;
            }
        }

        return max;
    }
}

File: medium/container-with-most-water.go
func maxArea(height []int) int {
    // The first submission -> 99ms
    l := 0;
    r := len(height) - 1;
    max := 0;
    curr := 0;

    for l < r {
        if height[l] < height[r] {
            curr = height[l] * (r - l);
            l++;
        } else {
            curr = height[r] * (r - l);
            r--;
        }
        if curr > max {
            max = curr;
        }
    }

    return max;
}